hawkular-command-gateway-itest/src/test/java/org/hawkular/cmdgw/ws/test/TestWebSocketClient.java

```diff
@@ -96,162 +190,383 @@ public boolean matches(ReusableBuffer actual, TestListener testListener) {
+            log.fine(msg);
+            return new MessageReport(new IllegalStateException(msg), actualEvent.getIndex());
+        }
+
+        public abstract MessageReport match(ActualEvent actualEvent);
 
-        public MessageReport assertMatch(TestListener testListener, int index, ReusableBuffer message,
-                                         PayloadType type) {
-            Description description = new StringDescription();
-            boolean fail = false;
-            if (!inMessageMatcher.matches(message, testListener)) {
-                description
-                        .appendText("Expected: ")
-                        .appendDescriptionOf(inMessageMatcher)
-                        .appendText("\n     but: ");
-                inMessageMatcher.describeMismatch(message, description);
-                fail = true;
-            }
-            if (!inTypeMatcher.matches(type)) {
-                description
-                        .appendText("Expected: ")
-                        .appendDescriptionOf(inTypeMatcher)
-                        .appendText("\n     but: ");
-                inTypeMatcher.describeMismatch(type, description);
-                fail = true;
-            }
-
-            if (fail) {
-                return new MessageReport(new AssertionError(description.toString()), index);
+
+        public void scheduleAnswer(ActualEvent actualEvent, ExecutorService executor, WebSocket webSocket) {
+            if (answer != null) {
+                answer.schedule(executor, webSocket);
             } else {
-                return MessageReport.passed(index);
+                log.fine("No answer to send for message[" + actualEvent.getIndex() + "]");
             }
         }
+    }
 
-        /**
-         * @return a binary attachment to send out with {@link #getTextAnswer()} if the incoming message matched the
-         * expectations
-         */
-        public URL getBinaryAnswer() {
-            return binaryAnswer;
+    public static class MessageAnswer implements Answer {
+        private final URL binaryAnswer;
+
+        private final long sleepAfterAnswerMs;
+        private final String textAnswer;
+        public MessageAnswer(String textAnswer) {
+            this(textAnswer, null, 0);
+        }
+        public MessageAnswer(String textAnswer, URL binaryAnswer, long sleepAfterAnswerMs) {
+            super();
+            this.textAnswer = textAnswer;
+            this.binaryAnswer = binaryAnswer;
+            this.sleepAfterAnswerMs = sleepAfterAnswerMs;
         }
 
         /**
-         * @return a text message to send out if the incoming message matched the expectations
+         * @param webSocket
+         * @return
          */
-        public String getTextAnswer() {
-            return textAnswer;
+        public void schedule(final ExecutorService executor, final WebSocket webSocket) {
+            executor.execute(new Runnable() {
+                @Override
```

Comment:
So far, I was not a friend of over-lambdaization ;) Yes, you are right that it is less lines.